Simplified/NYPLBookCellDelegate.m

```diff
@@ -175,70 +175,76 @@ - (void)openAudiobook:(NYPLBook *)book {
     dict[@"id"] = book.identifier;
   }
   
-  id<Audiobook> const audiobook = [AudiobookFactory audiobook: dict ?: json];
-
-  if (!audiobook) {
-    [self presentUnsupportedItemError];
-    return;
-  }
+  // Check vendor's DRM key and update it if necessary
+  [AudiobookCertificate updateVendorKeyWithBook:json completion:^{
```

Comment:
this is what I was referring to earlier -- the updateVendorKeyWithBook may fail, but no matter of success of failure we are doing the same thing here, which seems weird to me.

Reply:
added an error; if audiobook is not created (is nil) the error will be shown.